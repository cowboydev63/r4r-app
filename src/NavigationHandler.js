import React from 'react';
import { connect } from 'react-redux';
import { clearError } from './state/error/actions';
import { setFetchingStatus } from './state/api/actions';

class NavigationHandler extends React.Component{
    componentDidUpdate(prevProps){
        if(this.props.location !== prevProps.location){
            window.scrollTo(0,0);
            if(this.props.error){
                this.props.clearError();
            }
        }
    }

    render(){
        return this.props.children;
    }
}
const mapStateToProps = ({ error, router }) => ({
    error,
    location: router.location,
})

const mapDispatchToProps = {
    clearError,
    setFetchingStatus,
}

export default connect(mapStateToProps, mapDispatchToProps)(NavigationHandler);