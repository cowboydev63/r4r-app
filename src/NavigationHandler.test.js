import React from 'react';
import { shallow } from 'enzyme';
import { NavigationHandler } from './NavigationHandler';

it('renders without crashing', () => {
  shallow(<NavigationHandler />);
});