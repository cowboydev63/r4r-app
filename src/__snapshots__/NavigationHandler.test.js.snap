// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationHandler HOC renders correctly 1`] = `""`;
